frontend/src/metabase/modes/components/drill/ObjectDetailDrill.jsx

```diff
@@ -11,26 +10,14 @@ function hasManyPKColumns(question) {
 
 function getActionForPKColumn({ question, column, objectId, extraData }) {
   if (hasManyPKColumns(question)) {
-    // Filter by a clicked value, then a user can click on the 2nd, 3d, ..., Nth PK cells
-    // to narrow down filtering and eventually enter the object detail view once all PKs are filtered
     return ["question", () => question.filter("=", column, objectId)];
   }
 
   const isDashboard = !!extraData?.dashboard;
-  if (isDashboard) {
-    const { parameterValuesBySlug = {} } = extraData;
-    const hasParameters = Object.keys(parameterValuesBySlug).length > 0;
-
-    // This should result in a metabase/dashboard/actions navigateToNewCardFromDashboard call
-    // That will convert dashboard parameters into question filters
-    // and make sure the clicked row will be present in the query results
-    if (hasParameters) {
-      const getNextQuestion = () => question;
-      const getExtraData = () => ({ objectId });
-      return ["question", getNextQuestion, getExtraData];
-    }
 
-    return ["url", () => Urls.question(question.card(), { objectId })];
+  // the question from the dashboard may have filters applied already
+  if (isDashboard) {
+    return ["question", () => question];
   }
```

Comment:
@kulyk I think this is a simpler way to accomplish preserving dashboard filters when drilling through primary keys. Let me know if you think I'm missing something.

Reply:
my misgiving about the getExtraData approach is that it obscures what is going on down the chain and makes things hard to search for. I think we can the best of both worlds though by explicitly naming which extra props we're passing. I'll make the update.